js/client-battle.js

```diff
@@ -1083,6 +1112,202 @@
+				var min = 0;
+				var max = 0;
+				if (target.volatiles.formechange) template = Tools.getTemplate(target.volatiles.formechange[2]);
+				var minRatio = (pokemon.stats.spe / this.getTemplateMinSpeed(template, target.level));
+				var maxRatio = (pokemon.stats.spe / this.getTemplateMaxSpeed(template, target.level));
+				if (minRatio >= 4) min = 150;
+				else if (minRatio >= 3) min = 120;
+				else if (minRatio >= 2) min = 80;
+				else if (minRatio >= 1) min = 60;
+				else min = 40;
+				if (maxRatio >= 4) max = 150;
+				else if (maxRatio >= 3) max = 120;
+				else if (maxRatio >= 2) max = 80;
+				else if (maxRatio >= 1) max = 60;
+				else max = 40;
+				basePower = (min === max) ? max : min + ' to ' + max;
+			}
+			if (move.id === 'gyroball') {
+				var template = target;
+				if (target.volatiles.formechange) template = Tools.getTemplate(target.volatiles.formechange[2]);
+				var min = (Math.floor(25 * this.getTemplateMinSpeed(template, target.level) / pokemon.stats.spe) || 1);
+				var max = (Math.floor(25 * this.getTemplateMaxSpeed(template, target.level) / pokemon.stats.spe) || 1);
+				if (min > 150) min = 150;
+				if (max > 150) max = 150;
+				basePower = (min === max) ? max : min + ' to ' + max;
+			}
+			// Movements which have base power changed due to items.
+			if (pokemon.item && !this.battle.hasPseudoWeather('Magic Room') && !pokemon.volatiles['embargo']) {
+				if (move.id === 'fling') {
+					var item = Tools.getItem(pokemon.item);
+					if (item.fling) basePower = item.fling.basePower;
+				}
+				if (move.id === 'naturalgift') {
+					var item = Tools.getItem(pokemon.item);
+					if (item.naturalGift) basePower = item.naturalGift.basePower;
+				}
+			}
+			// Movements which have base power changed according to weight.
+			if (target.weightkg) {
+				var targetWeight = target.weightkg;
+				// Autotomize cannot be really known on client, so we calculate it's one charge.
+				if (target.volatiles.autotomize) targetWeight -= 100;
+				if (targetWeight < 0.1) targetWeight = 0.1;
+				if (move.id === 'lowkick' || move.id === 'grassknot') {
+					basePower = 20;
+					if (targetWeight >= 200) basePower = 120;
+					else if (targetWeight >= 100) basePower = 100;
+					else if (targetWeight >= 50) basePower = 80;
+					else if (targetWeight >= 25) basePower = 60;
+					else if (targetWeight >= 10) basePower = 40;
+					if (target.volatiles.autotomize) basePower += ' (Approximation)';
+				}
+				if (move.id === 'heavyslam' || move.id === 'heatcrash') {
+					basePower = 40;
+					if (pokemonWeight > targetWeight * 5) basePower = 120;
+					else if (pokemonWeight > targetWeight * 4) basePower = 100;
+					else if (pokemonWeight > targetWeight * 3) basePower = 80;
+					else if (pokemonWeight > targetWeight * 2) basePower = 60;
+					if (target.volatiles.autotomize) basePower += ' (Approximation)';
+				}
+			}
+			// Other ability boosts.
+			if (pokemon.ability === 'technician' && basePower <= 60) {
```

Comment:
It seems like `basePower` can be a string at this point, like `"45 to 47"` or `"80 (Approximation)""`. What happens if you compare such a string with `<= 60` or multiply it like `* 1.5`?